Spring Boot 4

olekvernberg opened the pull request.

olekvernberg committed:
Spring Boot 4

olekvernberg committed:
NaisProxyCustomizer skal være en component

olekvernberg committed:
Fix NaisProxyCustomizer

olekvernberg committed:
Fix timeout config

olekvernberg committed:
Bump major-versjon for å skille mellom spring 3 og 4

olekvernberg committed:
NaisProxyCustomizer må arve fra RestTemplateCustomizer

olekvernberg commented on http-client/src/main/java/no/nav/familie/http/config/NaisProxyCustomizer.kt:
```diff
@@ -10,33 +11,41 @@ import org.apache.hc.core5.http.io.SocketConfig
     @Value("\${familie.nais.proxy.connectTimeout:15000}") val connectTimeout: Long,
     @Value("\${familie.nais.proxy.socketTimeout:15000}") val socketTimeout: Long,
     @Value("\${familie.nais.proxy.requestTimeout:15000}") val requestTimeout: Long,
-) : INaisProxyCustomizer {
```
Good catch. Tenkte jeg ikke på. Jeg forsøkte meg på en fix

stigebil reviewed (approved):
Hvis hovedbranchen skal være springboot 4, så må man endre:
.github/workflows/build-deploy.yml
`export VERSION=4.${GIT_COMMIT_DATE}_${GIT_COMMIT_HASH}`

Da må man branche ut Spring boot 3.5.x ut i egen branch først og sørge for at man kan deploy 3 versjoner herfra.


Evt omvendt: Egen deploy for Spring 4

olekvernberg committed:
Deploy versjon 4

stigebil committed:
Bump til riktig token-support

stigebil committed:
Fiks etter rebase fra main

stigebil committed:
Kopiert http-client til rest-klient, som bruker jackson 3

Som en del av overgangen til Spring Boot 4, så bør man også gå over til jackson 3. Dette er en klone av http-client, men skrevet om til å bruke jackson 3 i stedet for jackson 2

objectMapper-> jsonMapper
com.fasterxml -> tools.jackson

stigebil committed:
Lagt til rest-klient i BOM

olekvernberg committed:
T kan være null og må skrives eksplisitt for returverdi i SB4

olekvernberg committed:
Rollback forrige commit

stigebil committed:
Oppgrader valutakurs-klient til å bruke ŕest-klient og Jackson 3 (#931)

stigebil committed:
Legg til ResponseBodyNullException for bedre håndtering av null-respons i REST-klient

stigebil committed:
Støtte for at ping kan ha null body

stigebil committed:
Skjul warn-logg på ResponseBodyNullException

stigebil committed:
Fjerner JsonMapper fra restklient
